src/Outsystems.SetupTools/Functions/Install-OSServerPreReqs.ps1

```diff
@@ -504,6 +514,89 @@ function Install-OSServerPreReqs
             }
         }
 
+        if ($mostRecentHostingBundleVersion -and $OnlyMostRecentHostingBundlePackage)
+        {
+            $isInstalled = IsDotNetCoreUninstallToolInstalled
+            if (-not $isInstalled)
+            {
+                try
+                {
+                    LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Installing .NET Uninstall Tool"
+                    $exitCode = InstallDotNetCoreUninstallTool -MajorVersion '1.5' -Sources $SourcePath
+                }
+                catch [System.IO.FileNotFoundException]
+                {
+                    LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Exception $_.Exception -Stream 3 -Message ".NET Uninstall Tool installer not found"
+                    WriteNonTerminalError -Message ".NET Uninstall Tool installer not found"
+
+                    $installResult.Success = $false
+                    $installResult.ExitCode = -1
+                    $installResult.Message = '.NET Uninstall Tool installer not found'
+
+                    return $installResult
+                }
+                catch
+                {
+                    LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Exception $_.Exception -Stream 3 -Message "Error downloading or starting the .NET Uninstall Tool installation"
+                    WriteNonTerminalError -Message "Error downloading or starting the .NET Uninstall Tool installation"
+
+                    $installResult.Success = $false
+                    $installResult.ExitCode = -1
+                    $installResult.Message = 'Error downloading or starting the .NET Uninstall Tool installation'
+
+                    return $installResult
+                }
+            }
+            else
+            {
+                LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message ".NET Uninstall Tool found"
+            }
+
+            $env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path","User")
+
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Uninstalling previous ASP.NET Core Runtime packages"
+            $isUninstalled = UninstallPreviousDotNetCorePackages -Package '--aspnet-runtime' -Version $mostRecentHostingBundleVersion
+            if (-not $isUninstalled)
+            {
+                LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Exception $_.Exception -Stream 3 -Message "Error uninstalling previous ASP.NET Core Runtime packages"
+                WriteNonTerminalError -Message "Error uninstalling previous ASP.NET Core Runtime packages"
+
+                $installResult.Success = $false
+                $installResult.ExitCode = -1
+                $installResult.Message = 'Error uninstalling previous ASP.NET Core Runtime packages'
+
+                return $installResult
+            }
```

Comment:
This piece of code repeats 3 times for different packages, we could refactor this

Reply:
Refactored it 👌